docs/2.9.0/docs/query/pqs-user-guide.rst

```diff
@@ -1,466 +1,1741 @@
+           Suppressed: java.net.ConnectException: Connection refused application=scribe
+
+Metrics
+~~~~~~~
+
+The following metrics are available to monitor stability of PQS's
+dependencies. See the `observability <#observability>`__ section for
+more details on general observability:
+
+.. code:: text
+
+   # TYPE app_restarts_total counter
+   # HELP app_restarts_total Number of total app restarts due to recoverable errors
+   app_restarts_total{,exception="Recoverable GRPC exception."} 5.0
+
+   # TYPE grpc_up gauge
+   # HELP grpc_up Grpc channel is up
+   grpc_up{} 1.0
+
+   # TYPE jdbc_conn_pool_up gauge
+   # HELP jdbc_conn_pool_up JDBC connection pool is up
+   jdbc_conn_pool_up{} 1.0
+
+Instability counters reset
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Once PQS starts experiencing a period of instability, it starts
+maintaining (and increasing) necessary counters during each retry
+attempt. These counters are reset only after a period of continuous
+stability, as defined by ``--retry-counter-reset``. As such, during the
+prolonged periods of intermittent failures that alternate with brief
+periods of operating normally, PQS keeps maintaining a cautious stance
+on assumptions regarding the stability of the overall system. This can
+be illustrated with an example below:
+
+.. code:: text
+
+   # --retry-counter-reset=PT5M
+
+   time -->       1:00            5:00               10:00
+                   v               v                   v
+   operation:  ====xx=x====x=======x========================
+                   ^               ^                   ^
+                   A               B                   C
+
+   x - a failure causing retry happens
+   = - operating normally
+
+In the timeline above, intermittent failures start at point A, and each
+retry attempt contributes to the increase of the overall backoff
+schedule. Consequently, each subsequent retry allows more time for the
+system to recover. This schedule does not reset to its initial values
+until after the configured period of stability is reached following the
+last failure (point B), such as after operating without any failures for
+5 minutes (point C).
+
+Exit codes
+~~~~~~~~~~
+
+PQS terminates with the following exit codes:
+
+-  ``0``: Normal termination
+-  ``1``: Termination due to unrecoverable error or all retry attempts
+   for recoverable errors have been exhausted
+
+.. _ledger-streaming--recovery:
+
+Ledger Streaming & Recovery
+---------------------------
+
+On (re-)start, PQS determines last saved checkpoint and continues
+incremental processing from that point onward. PQS is able to start and
+finish at prescribed ledger offsets, specified via args:
+
+.. code:: shell
+
+   $ ./scribe.jar pipeline --help-verbose
+
+   Usage: pipeline SOURCE TARGET [OPTIONS]
+   [...]
+   Options:
+   [...]
+     --pipeline-ledger-start [enum | string]    Start offset (default: Latest)
```

Comment:
By "string", do you mean the offset number?

Reply:
It's not a number - it's actually a string on the Ledger API. We just happen to have a convention that it seems to be rendered as a hexademical, but we treat it as a string and we can't read anything into it other than it sorts.

Reply:
(this gets fixed in Canton 3.1 where it will become a number)